_posts/en/newsletters/2021-01-13-newsletter.md

```diff
@@ -0,0 +1,159 @@
+---
+title: 'Bitcoin Optech Newsletter #131'
+permalink: /en/newsletters/2021/01/13/
+name: 2021-01-13-newsletter
+slug: 2021-01-13-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter describes a new proposed Bitcoin P2P protocol
+message, a BIP for the bech32 modified address format, and an idea for
+preventing UTXO probing in proposed dual-funded LN channels.  Also
+included are our regular sections with the summary of a Bitcoin Core PR
+Review Club meeting, a list of releases and release candidates, and
+descriptions of notable changes to popular Bitcoin infrastructure
+software.
+
+## News
+
+- **Proposed `disabletx` message:** in 2012, [BIP37][] was published,
+  giving lightweight clients the ability to request that peers not
+  relay to the client any unconfirmed transactions until the client
+  had loaded a [bloom filter][topic transaction bloom filtering].
+  Bitcoin Core [later][bitcoin core #6993] reused this mechanism for its
+  bandwidth-saving `-blocksonly` mode where a node requests that its
+  peers don't send it any unconfirmed transactions at all.  Last year,
+  Bitcoin Core with default settings began opening a few
+  block-relay-only connections as an efficient way to improve [eclipse
+  attack][topic eclipse attacks] resistance without significantly
+  increasing bandwidth or reducing privacy (see [Newsletter #63][news63
+  bcc15759]).  However, the BIP37 mechanism being used to suppress
+  transaction relay allows the initiating node to request full
+  transaction relay at any time.  Transaction relay can consume hundreds
+  of megabytes of data per day per peer, so nodes need to set their anti
+  Denial of Service (DoS) limits with the assumption that each
+  BIP37-based low-bandwidth blocks-relay-only connection could suddenly
+  become a full transaction relay connection.
```

Comment:
I'm not sure what you mean by "anti Denial of Service limits" here. The resource issue is that  transaction-relaying peers need to have the `CNode::TxRelay` structure initialized, which contains bloom filters and sets of tx hashes, so can use quite a bit of memory. We need to initialize this structure for all peers which may eventually start relaying transactions, which currently includes all inbound peers.

Reply:
I didn't know about that, thanks.  I'll update the text.  I assumed the main issue was tx inv overhead, which is basically wasted bandwidth if more connections are opened than necessary.